security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,207 @@
+  - Not applicable to OP
+  - Requests sent to this contract are ignored, and the contract is not deployed by default
+- EIP-6110: Supply validator deposits on chain
+  - Not applicable to OP
+  - Requests sent to this contract are ignored, and the contract is not deployed by default
+- EIP-7685: General purpose execution layer requests
+  - No requests are currently applicable to OP, so the [execution requests array is always empty in Isthmus](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/exec-engine.md#engine_newpayloadv4).
+
+## Failure Modes and Recovery Paths
+
+### FM1: Chain halt due to upgrade transaction failing
+
+- **Description:** The chain could halt if a network upgrade transaction can't be included in the upgrade block.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. end-to-end tests created for network upgrade transactions
+      - [Action test created here](https://github.com/ethereum-optimism/optimism/blob/01ddb2e6a09edf55a7cb2130e0a5b6acd0c2d2fa/op-e2e/actions/upgrades/isthmus_fork_test.go#L290)
+  2. same upgrade path as previous predeploys
+      - example: [EIP-4788 deployment transaction](https://github.com/ethereum-optimism/optimism/blob/b6b74290b5502f45daae57946f969327cdb2d383/op-node/rollup/derive/ecotone_upgrade_transactions.go#L130)
+    
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing the network upgrade transaction in rollup node client software (like `op-node`).
+
+### FM2: Chain halt due to request hash mismatch
+
+- **Description:** Requests hash was added to the block header. If it mismatches between the EL client and the rollup node, the chain could halt.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. End-to-end test ensuring no requests are included in blocks
+      - https://github.com/ethereum-optimism/optimism/pull/14253
+  2. Simplicity of validation rules - always must be an empty hash
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing requests hash logic in respective clients (e.g. `op-node`, `op-geth`, `op-reth`, ...)
+
+### FM3: EIP-7702 transactions cannot be included on the chain
+
+- **Description:** A client implementation could have a bug that disallows SetCode (EIP-7702) transactions from being included on chain. This could cause an unintended fork by the affected client.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:** 
+	1. End-to-end tests of set code transactions before and after the fork
+        - https://github.com/ethereum-optimism/optimism/pull/14197
+        - https://github.com/ethereum-optimism/optimism/pull/14288
+- **Detection:** Sending set code transactions would fail - manual detection is most likely.
```

Comment:
Can you detail more about the *manual* detection? 
You mean the user will see his transaction reverting directly or *not supported*. 
But, would be a case to case stuff?

I wondering if would make sense to you guys, to craft a small `eth_call` with a type 4 that we can run against any EL on L2 to detect if this one is updated or not.

Reply:
Depends on the signer used I think. It can't happen until Isthmus is activated. Blocks and tx pool will reject until Isthmus is activated (at least for Geth): https://github.com/ethereum-optimism/op-geth/pull/535/files#diff-58204a02692cd9ef1cf0767b18dbc4df99eb678126c8982350e5b727af075b5eR74

If a client accidentally does accept them early and include them in a block, it would cause an L2 chain halt for the non-sequencer client.

If a non-sequencer client does not accept `type 4` txs after they should be supported, it would also cause an L2 chain halt for the non-sequencer client.

Reply:
Thank you for clarifying!
Make sense here.